Add page protection

vgeorge opened the pull request:
This adds next-auth middleware to protect pages that require authentication. Contributes to #95.

Changes included:

- Added next-auth middleware
- Add Cypress spec to check all protected pages
- Added NEXTAUTH_SECRET to the development environment
- Added Cypress Eslint recommended settings

How to give feedback:

- Start the app
- Visit /profile unauthenticated, it should display the sign-in page
- Visit /profile authenticated, it should display the profile page
- Try visiting other protected routes

Other notes:

- The sign-in page is auto-generated by next-auth, we can [add a custom one](https://next-auth.js.org/configuration/options#pages) if needed.

@LanesGood this is ready for review.

LanesGood commented:
@vgeorge It looks good for the profile page. I also see that `/teams` is redirecting to the sign-in page. I don't think this is the intention. The "Explore all teams" page should be public; we have a control to allow public vs private views of teams.

vgeorge commented:
Status update:

-  Removed `/teams/*` page from the general auth middleware and the pages should now be public
- Moved Team "model" file form `/app/lib` to `/src/models`. This folder name convention provides more clarity on the contents of the file. We also need to move the organization and profile files in `/app/lib` to `/src/models`, but that can be done in a separate PR to avoid multiple changes
- Added a "TESTING=true" flag to avoid hitting the OSM API when resolving names on testing environments. I didn't use `NODE_ENV` because we run the development server for E2E and Next.js do not keep the value of `NODE_ENV` set to `test` on server reloads, even when passed in the command line
- Added a Cypress task to clear and populate the database for testing

@LanesGood thanks for the review and catching the regression. I'm assigning @kamicut for the next review because there are important changes in the server code. This is ready for review.

kamicut reviewed (approved):
LGTM
